mobile/src/main/java/ch/epfl/sdp/mobile/ui/setting/EditProfileImageDialog.kt

```diff
@@ -0,0 +1,127 @@
+package ch.epfl.sdp.mobile.ui.setting
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.application.Profile.Color
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.state.toColor
+import ch.epfl.sdp.mobile.ui.PawniesColors.Green800
+import ch.epfl.sdp.mobile.ui.prepare_game.Dialog
+
+/**
+ * Component for display a Dialog to edit the profile image
+ *
+ * @param state the [EditProfileImageDialogState] as an argument.
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+fun EditProfileImageDialog(
+    state: EditProfileImageDialogState,
+    modifier: Modifier = Modifier,
+) {
+
+  /*
+   A bug in Compose's navigation component makes the system window shrink to the measured size of
+   the dialog when it's filled for the first time. On the following recompositions, this new size
+   is applied as the constraints to the root of the hierarchy and some elements might not be able
+   to occupy some space they need.
+   Applying Modifier.fillMaxSize() makes sure we "reserve" this space and that the window will
+   never force us to shrink our content.
+  */
+  Box(modifier.fillMaxSize(), Alignment.Center) {
+    val strings = LocalLocalizedStrings.current
+
+    Dialog(
+        cancelContent = { Text(text = strings.settingEditCancel) },
+        confirmContent = { Text(text = strings.settingEditSave) },
+        onCancelClick = state::onCancelClick,
+        onConfirmClick = state::onSaveClick) {
+      Column(modifier = Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
+        EditSettingPicture(backgroundColor = state.backgroundColor, emoji = state.emoji)
+        Text(strings.settingProfileNameLabel, style = MaterialTheme.typography.subtitle1)
+        LazyRow() { items(items = emojis) { item -> SelectEmojiItem(state, item) } }
+        LazyRow() {
+          items(items = Color.asList()) { item -> SelectBackgroundColorItem(state, item) }
+        }
```

Comment:
I suggest not applying the `Modifier.padding` to the top-level column, but to inner composables. This way, you could fix the clipping of the scrollable content.

## Suggested fix (which you may copy-paste)

```kotlin
Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Spacer(Modifier.height(16.dp))
        EditSettingPicture(
            backgroundColor = state.backgroundColor,
            emoji = state.emoji,
            modifier = Modifier.padding(horizontal = 16.dp),
        )
        Text(
            strings.settingProfileImageLabel,
            style = MaterialTheme.typography.subtitle1,
            modifier = Modifier.padding(horizontal = 16.dp),
        )
        LazyRow(contentPadding = PaddingValues(horizontal = 16.dp)) {
          items(items = emojis) { item -> SelectEmojiItem(state, item) }
        }
        LazyRow(contentPadding = PaddingValues(horizontal = 16.dp)) {
          items(items = Color.asList()) { item -> SelectBackgroundColorItem(state, item) }
        }
        Spacer(Modifier.height(16.dp))
      }
```

## (current) Broken clipping

![telegram-cloud-photo-size-4-6019269865480763295-y](https://user-images.githubusercontent.com/6318990/166456578-43d85671-0b14-4d21-a0f5-7f01d462f32c.jpg)

## (provided code) Fixed clipping, but with the same visual paddings

![telegram-cloud-photo-size-4-6019269865480763297-y](https://user-images.githubusercontent.com/6318990/166456629-3d4f7dc6-a818-4cef-bc4a-6270b9c3d07a.jpg)

Reply:
has been changed